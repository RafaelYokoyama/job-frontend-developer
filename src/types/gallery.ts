export interface Attraction {
  images: { url: string }[];
}

export interface GalleryProps {
  attractions: Attraction[];
}
