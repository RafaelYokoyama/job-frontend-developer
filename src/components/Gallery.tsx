import React from 'react';
import Image from 'next/image';
import { GalleryProps } from 'types/gallery';

const Gallery = ({ attractions }: GalleryProps) => {
  return (
    <div>
      {attractions.map((attraction, attractionIndex) => (
        <div key={attractionIndex} className="mt-4">
          <h2 className="text-lg font-semibold text-gray-800 mb-2">
            Galeria de Imagens {attractionIndex + 1}
          </h2>
          <div className="grid grid-cols-8 gap-1">
            {attraction.images.map((image, imageIndex) => (
              <Image
                key={imageIndex}
                className="w-[38px]  h-[38px] object-cover rounded-lg"
                width={38}
                height={38}
                src={image.url}
                alt={`Image ${imageIndex + 1} of Gallery ${
                  attractionIndex + 1
                }`}
              />
            ))}
          </div>
        </div>
      ))}
    </div>
  );
};

export default Gallery;
