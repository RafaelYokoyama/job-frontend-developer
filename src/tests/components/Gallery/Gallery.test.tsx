import React from 'react';
import { render } from '@testing-library/react';
import Gallery from 'components/Gallery';
import { mockGallery } from 'tests/mocks/gallery';

describe('Gallery Component', () => {
  it('checks a snapshot', () => {
    const { container } = render(<Gallery attractions={mockGallery} />);
    expect(container).toMatchSnapshot();
  });
  it('should render gallery with correct number of images', () => {
    const { getAllByAltText } = render(<Gallery attractions={mockGallery} />);
    const images = getAllByAltText(/Image/);
    expect(images.length).toBe(3);
  });

  it('should render gallery headings correctly', () => {
    const { getByText } = render(<Gallery attractions={mockGallery} />);
    mockGallery.forEach((attraction, index) => {
      const heading = getByText(`Galeria de Imagens ${index + 1}`);
      expect(heading).toBeInTheDocument();
    });
  });
});
