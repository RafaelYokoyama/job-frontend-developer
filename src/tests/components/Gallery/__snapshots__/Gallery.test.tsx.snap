// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Component checks a snapshot 1`] = `
<div>
  <div>
    <div
      class="mt-4"
    >
      <h2
        class="text-lg font-semibold text-gray-800 mb-2"
      >
        Galeria de Imagens 
        1
      </h2>
      <div
        class="grid grid-cols-8 gap-1"
      >
        <img
          alt="Image 1 of Gallery 1"
          class="w-[38px]  h-[38px] object-cover rounded-lg"
          data-nimg="1"
          decoding="async"
          height="38"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage1.jpg&w=96&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage1.jpg&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage1.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="38"
        />
        <img
          alt="Image 2 of Gallery 1"
          class="w-[38px]  h-[38px] object-cover rounded-lg"
          data-nimg="1"
          decoding="async"
          height="38"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage2.jpg&w=96&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage2.jpg&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage2.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="38"
        />
      </div>
    </div>
    <div
      class="mt-4"
    >
      <h2
        class="text-lg font-semibold text-gray-800 mb-2"
      >
        Galeria de Imagens 
        2
      </h2>
      <div
        class="grid grid-cols-8 gap-1"
      >
        <img
          alt="Image 1 of Gallery 2"
          class="w-[38px]  h-[38px] object-cover rounded-lg"
          data-nimg="1"
          decoding="async"
          height="38"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage3.jpg&w=96&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage3.jpg&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage3.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="38"
        />
      </div>
    </div>
  </div>
</div>
`;
